as-ti-control/as_ti_control/ll_classes.py

```diff
@@ -1,144 +1,135 @@
+EVEs = Connections.get_devices('EVE')
+AFCs = Connections.get_devices('AFC')
+FOUTs = Connections.get_devices('FOUT')
+TWDS_EVG = Connections.get_connections_twds_evg()
 
 
-class _Timer(_Thread):
-    def __init__(self, interval, function, args=tuple(), niter=100):
-        super().__init__(daemon=True)
-        self.interval = interval
-        self.function = function
-        self.args = args
-        self.niters = niter
-        self.cur_iter = 0
-        self.stopped = _Event()
+class _Base:
 
-    def run(self):
-        self.function(*self.args)
-        while ((not self.stopped.wait(self.interval)) and
-               self.niters > self.cur_iter):
-            self.cur_iter += 1
-            self.function(*self.args)
-
-    def reset(self):
-        self.cur_iter = 0
-
-    def stop(self):
-        self.stopped.set()
-
-
-def get_ll_trigger_object(channel, callback, init_state, evg_params):
-    """Get Low Level trigger objects."""
-    LL_TRIGGER_CLASSES = {
-        ('EVR', 'OUT'): _LL_TrigEVROUT,
-        ('EVR', 'OTP'): _LL_TrigEVROTP,
-        ('EVE', 'OUT'): _LL_TrigEVEOUT,
-        ('AFC', 'CRT'): _LL_TrigAFCCRT,
-        ('AFC', 'FMC'): _LL_TrigAFCFMC,
-        }
-    chan = _PVName(channel)
-    conn_ty, conn_num = IOs.LL_RGX.findall(chan.propty)[0]
-    key = (chan.dev, conn_ty)
-    cls_ = LL_TRIGGER_CLASSES.get(key)
-    if not cls_:
-        raise Exception('Low Level Trigger Class not defined for device ' +
-                        'type '+key[0]+' and connection type '+key[1]+'.')
-    return cls_(channel, int(conn_num), callback, init_state, evg_params)
-
-
-class _LL_Base:
-    def __init__(self, callback, init_state):
+    def __init__(self, callback, init_hl_state):
         """Initialize the Low Level object.
 
         callback is the callable to be called each time a low level PV changes
         its value.
-        init_state is the initial value of the high level properties.
+        init_hl_state is the initial value of the high level properties.
         """
-        self._dict_functions_for_write = self._get_dict_for_write()
-        self._dict_functions_for_read = self._get_dict_for_read()
-        self._dict_convert_prop2pvrb = self._get_convertion_prop2pvrb()
-        self._dict_convert_pvrb2prop = {
-                val: key for key, val in self._dict_convert_prop2pvrb.items()}
+        self._dict_functions_for_write = self._define_dict_for_write()
+        self._dict_functions_for_read = self._define_dict_for_read()
+        self._dict_convert_prop2pv = self._define_convertion_prop2pv()
+        self._dict_convert_pv2prop = {
+                val: key for key, val in self._dict_convert_prop2pv.items()}
         self.callback = callback
-        self._hl_state = init_state
-        self._ll_state = dict()
+        self._my_state_sp = dict()
         self._rf_freq = RFFREQ
         self._rf_div = RFDIV
+
         self._rf_freq_pv = _epics.PV(LL_PREFIX + 'SI-03SP:RF-SRFCav:Freq-SP',
```

Comment:
I just have a doubt about the PV name of RF frequency. I think it must be independent of the cavity device.

Reply:
Thank  you, @anacso17! I will make a commit to correct it.